### PR TITLE
Optimise Dilate() function performance

### LINKED ISSUE #463: Optimise Dilate() function performance
We have **Dilate()** function in `src/image_function.cpp` under unnamed namespace. There are places where we could potentially speed up function's execution:
1. after reaching the condition `if( imageXEnd - imageX < dilateX )` or `if( height - rowId < dilationY )` we should stop loop as we reached end of line/column.
2. Initialize `const uint32_t rowSize = image.rowSize();` only once
3. Add check if `dilationX` and `dilationY` are equal to 0 then exit function

Additionally, as a part of future commit please remove one empty line before `if( dilationX > width / 2 )`.

### PR DESCRIPTION
**Pull request by theoniko**
Closes #463

### CODE REVIEW

File: src/image_function.cpp [hunk 1]
```diff
@@ -58,19 +58,23 @@ namespace
         Image_Function::ParameterValidation( image, x, y, width, height );
         Image_Function::VerifyGrayScaleImage( image );
 
+	 if (dilationX == 0u && dilationY == 0u) {
```
Comment (ihhub) [inline src/image_function.cpp]: Please do not use `{` and `}` for single line code inside if block to make code look nicer :)

File: src/image_function.cpp [hunk 2]
```diff
@@ -90,10 +93,13 @@ namespace
                         else
                             startPos[pairCount] = imageX - dilateX;
 
-                        if( imageXEnd - imageX < dilateX )
-                            endPos[pairCount] = imageXEnd;
-                        else
+                        if ( imageXEnd - imageX < dilateX ) {
+			     endPos[pairCount] = imageXEnd;
+			     break;
```
Comment (ihhub) [inline src/image_function.cpp]: We have to increase pairCount before break.

File: src/image_function.cpp [hunk 3]
```diff
@@ -58,19 +58,22 @@ namespace
         Image_Function::ParameterValidation( image, x, y, width, height );
         Image_Function::VerifyGrayScaleImage( image );
 
-
+	 if ( dilationX == 0u && dilationY == 0u )
```
Comment (ihhub) [inline src/image_function.cpp]: @theoniko could you please check indentation of lines. Here we have 9 spaces instead of 8 :)

**Comment by theoniko** [comment]
Hello @ihhub,

Thank you for your review. I will make the requested changes once i have free time.

**Comment by ihhub** [comment]
@theoniko just a small note: please use 4 spaces instead of tabs according to your coding style :)

**Comment by ihhub** [comment]
@theoniko thank you very much for your continuous support of the project!

**Pull request merged**